pepdbagent/modules/user.py

```diff
@@ -0,0 +1,196 @@
+
+        :param namespace: user namespace
+        :return: user id
+        """
+        new_user_raw = User(namespace=namespace)
+
+        with Session(self._sa_engine) as session:
+            session.add(new_user_raw)
+            session.commit()
+            user_id = new_user_raw.id
+        return user_id
+
+    def get_user_id(self, namespace: str) -> Union[int, None]:
+        """
+        Get user id using username
+
+        :param namespace: user namespace
+        :return: user id
+        """
+        statement = select(User.id).where(User.namespace == namespace)
+        with Session(self._sa_engine) as session:
+            result = session.execute(statement).one_or_none()
+
+        if result:
+            return result[0]
+        return None
+
+    def add_to_favorites(self, namespace, project_namespace, project_name, project_tag) -> None:
+        """
+        Add project to favorites
+
+        :param namespace: namespace of the user
+        :param project_namespace: namespace of the project
+        :param project_name: name of the project
+        :param project_tag: tag of the project
+        :return: None
+        """
+
+        project_id = PEPDatabaseProject(self._pep_db_engine).get_project_id(
+            project_namespace, project_name, project_tag
+        )
+        user_id = self.get_user_id(namespace)
+
+        if not user_id:
+            user_id = self.create_user(namespace)
+
+        new_favorites_raw = Stars(user_id=user_id, project_id=project_id)
+        try:
+            with Session(self._sa_engine) as session:
+                session.add(new_favorites_raw)
+                session.commit()
+        except IntegrityError:
+            raise ProjectAlreadyInFavorites()
+        return None
+
+    def remove_from_favorites(
+        self, namespace: str, project_namespace: str, project_name: str, project_tag: str
+    ) -> None:
+        """
+        Remove project from favorites
+
+        :param namespace: namespace of the user
+        :param project_namespace: namespace of the project
+        :param project_name: name of the project
+        :param project_tag: tag of the project
+        :return: None
+        """
+        _LOGGER.debug(
+            f"Removing project {project_namespace}/{project_name}:{project_tag} from fProjectNotInFavorites for user {namespace}"
+        )
+        project_id = PEPDatabaseProject(self._pep_db_engine).get_project_id(
+            project_namespace, project_name, project_tag
+        )
+        user_id = self.get_user_id(namespace)
+        statement = delete(Stars).where(
+            and_(
+                Stars.user_id == user_id,
+                Stars.project_id == project_id,
+            )
+        )
+
+        with Session(self._sa_engine) as session:
+            result = session.execute(statement)
+            session.commit()
+            row_count = result.rowcount
+        if row_count == 0:
+            raise ProjectNotInFavorites(
```

Comment:
Same here, does this need to be an exception? If I don't have a project starred, and I say "remove this from my stars", can we just say "ok, its not starred"?